app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/CompareMenusBotViewModel.kt

```diff
@@ -63,16 +65,24 @@ class CompareMenusBotViewModel @Inject constructor(
                 is EateryApiResponse.Success -> {
                     _compareMenusUiState.update { currentState ->
                         currentState.copy(
-                            eateries = eateriesApiResponse.data.filter { eatery ->
-                                filters.all { filter ->
-                                    filter.passesFilter(
-                                        eatery,
-                                        favorites,
-                                        selected
-                                    )
+                            eateries = if (firstEatery != null) {
+                                (listOf(firstEatery!!) + (eateriesApiResponse.data.filter { eatery ->
+                                    eatery.name != firstEatery!!.name && filters.all { filter ->
+                                        filter.passesFilter(eatery, favorites, selected)
+                                    }
+                                }))
+                            } else {
+                                eateriesApiResponse.data.filter { eatery ->
+                                    filters.all { filter ->
+                                        filter.passesFilter(eatery, favorites, selected)
+                                    }
                                 }
                             },
```

Comment:
This code's a bit strange to me. I saw that you called:

```
selectedEateriesFlow.update {
    it + eatery
}
```

below upon initializing first eatery.

Why then do we need to do this `firstEatery` stuff? does it not work without it?

If all this code is doing is basically just putting the first eatery up top, we can likely make it simpler with:

`listOfNotNull(firstEatery)` and `eatery.name != firstEatery?.name`

I'd prefer not having this really fat if else block in the logic here so these just make the if/else logic sort of inlined with null operators.

Also, if REALLY all this is doing is just moving the first eatery up to the top, you can almost definitely do an EVEN simpler approach with `sortedWith` with a comparator that just says that the first eatery should come before any other eatery.

Reply:
thank you, just fixed! I thought about doing sortedWith but the approach i thought of (defining a custom comparator that needs to check firstEatery then sort by alphabetically then by if its open) seems to be more work. Lmk if it still looks lengthy